### ISSUE TITLE
How to embed a pre-trained tree into XGBRegressor?

### ISSUE THREAD

**Issue by ddghjikle** [dmlc/xgboost#7212]
Hi, during conducting transfer learning by XGBRegressor, i faced a question: how to embed weights of a pre-trained tree into XGBRegressor. Of course, I am also confused which tree is used in XGBRegressor?

**Comment by trivialfis** [comment]
Maybe:

``` python
reg.fit(X, y, xgb_mode=previous_model)
```

> Of course, I am also confused which tree is used in XGBRegressor?

It's not clear to me what's your question.  All the trees you have built by using `fit` are used in regressor.

**Comment by ddghjikle** [comment]
Hi, thanks very much for your help. I am trying to achieve transfer learning (with two datasets A and B) via XGBoost. To this end, I choose to train XGBoost on A and train a decision tree on B. After obtaining the XGBoost and decision tree, I want to embed weights of the decision tree into XGBoost, but i don't know how to achieve this embedding.

**Comment by ddghjikle** [comment]
Of course, I am also curious about other ideas of using XGBoost to achieve transfer learning.

**Comment by trivialfis** [comment]
So, you have a decision tree model trained using other libraries, and you want to build a gradient boosted model using XGBoost based on this existing decision tree model?  In this case, can you look into the `base_margin` parameter in `fit` method?

**Comment by trivialfis** [comment]
For a simple case, see https://github.com/dmlc/xgboost/blob/master/demo/guide-python/boost_from_prediction.py .  It's not using the sklearn interface, but can be a good starting point.

**Comment by ddghjikle** [comment]
Thanks. This is pretty helpful for me. By the way, are there any functions to print architecture or weights of trained XGBoost?

**Comment by trivialfis** [comment]
You can call `get_dump` or `save_model(model.json)`. The former is more human-readable and you can use dot format for plotting.
